src/state.ts

```diff
@@ -305,6 +307,14 @@ export default class State {
         return guildId in this._privilegedRolesByGuild;
     }
 
+    hasGuildSettings(guildId: string): boolean {
+        return guildId in this._settingsByGuild;
+    }
+
+    setGuildSettings(guildId: string, settings: Partial<Record<GuildSetting, number>>): void {
```

Comment:
For sake of ease, let's just define a type like `GuildSettings` or `GuildSettingMap` to refer to `Partial<Record<GuildSetting, number>>`